server/src/main/java/com/objectcomputing/checkins/services/team/TeamController.java

```diff
@@ -0,0 +1,94 @@
+package com.objectcomputing.checkins.services.team;
+
+import java.net.URI;
+import java.util.List;
+import java.util.UUID;
+
+import javax.annotation.Nullable;
+import javax.validation.Valid;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.annotation.Put;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.micronaut.http.annotation.Produces;
+
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+
+@Controller("/services/team")
+@Secured(SecurityRule.IS_ANONYMOUS)
+// @Secured(SecurityRule.IS_AUTHENTICATED)
+@Produces(MediaType.APPLICATION_JSON)
+@Tag(name="team")
+public class TeamController {
+
+    protected final TeamRepository teamRepository;
+
+    public TeamController(TeamRepository teamRepository){
+        this.teamRepository = teamRepository;
+    }
+    /**
+     * Find Team by name or find all.
+     * @param name
+     * @return
+     */
+    @Get("/{?name}")
+    public List<Team> findByValue(@Nullable String name) {
+
+        if(name != null) {
+            return teamRepository.findByName(name);
+        } else {
+            return teamRepository.findAll();
+        }
+    }
+
+    /**
+     * Save a new team.
+     * @param team
+     * @return
+     */
+    @Post("/")
+    // @Secured("VIEW")
+
+    public HttpResponse<Team> save(@Body @Valid Team team) {
+        Team newTeam = teamRepository.save(team);
+        
+        return HttpResponse
+                .created(newTeam)
+                .headers(headers -> headers.location(location(newTeam.getUuid())));
+    }
+
+    /**
+     * Update a Team.
+     * @param team
+     * @return
+     */
+    @Put("/")
+    public HttpResponse<?> update(@Body @Valid Team team) {
+
+        if(null != team.getUuid()) {
+            Team updatedTeam = teamRepository.update(team);
+            return HttpResponse
+                    .ok()
+                    .headers(headers -> headers.location(location(updatedTeam.getUuid())))
+                    .body(updatedTeam);
+                    
+        }
+        
+        return HttpResponse.badRequest();
+    }
+
+    protected URI location(UUID uuid) {
+        return URI.create("/services/team/" + uuid);
```

Comment:
Since this is hardcoded and is the same as the `@Controller` mapping, can we make a constant up top that gets used in both?

Reply:
https://github.com/oci-labs/check-ins/blob/feature-84/guilds/server/src/main/java/com/objectcomputing/checkins/services/guild/GuildController.java#L54

If you do it like this it will be dynamic to the route, no need for a constant.

Reply:
+1